bundles/org.openhab.binding.buienradar/src/main/java/org/openhab/binding/buienradar/internal/BuienradarHandler.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.buienradar.internal;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+
+import javax.measure.Unit;
+import javax.measure.quantity.Speed;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.buienradar.internal.buienradarapi.BuienradarPredictionAPI;
+import org.openhab.binding.buienradar.internal.buienradarapi.Prediction;
+import org.openhab.binding.buienradar.internal.buienradarapi.PredictionAPI;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tec.uom.se.unit.Units;
+
+/**
+ * The {@link BuienradarHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Edwin de Jong - Initial contribution
+ */
+@NonNullByDefault
+public class BuienradarHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(BuienradarHandler.class);
+
+    public static final Unit<Speed> MM_PER_HOUR = Units.METRE.divide(1000).divide(Units.HOUR).asType(Speed.class);
```

Comment:
Somehow I missed that. Got directed to SIUnits. The units system is quite opaque for a beginning Openhab developer (and this is from someone with extensive knowledge of a similar system in Scala). I'd expect to be able to *need* to write it this way (in terms of base units), since otherwise you get an explosion of different units.

Reply:
It seems this was added in the master branch, but I'm running this binding on 2.5.0M1. This would mean I can't run my own binding on my Odroid machine.

Reply:
No need to change then. It is absolutely fine to only use API of a released core version.

Reply:
It would only mean you can't run the binding compiled with the sources in what will be snapshot version. Nobody forbids you to locally modify the source to use the unit locally, compile it and run it. And even if it's kept anybody can put in a pr with this changed to SmarthomeUnits and than it will be modified anyway. So I see no reason why it should not be modified in this pr.

Reply:
Don't want to argue, but at one point you'd expect a stable base API, with a certain amount of backward and forward compatibility. I mean, no offense, but I spent a lot of my spare time writing this binding, searching again and again to do things 'the right way': setting up an IDE I don't know well (and frankly runs very slowly on my Mac), going through many CR comments, update my PR because I know I'm a noob in the ecosystem. Why? Because I want to do sth back to the ecosystem, having run OpenHAB for at least 4 years. You're friendly and all, don't get me wrong, and I understand you want to get things clean and of high quality, so would I want to have it as well.

However, the actual logic of this PR is around 30 lines. I wrote it in Scala as well, which is literally 7 lines including configuration. For you, writing such a binding would take 10 minutes, but for someone new to the ecosystem it takes 3-4 evenings, plus time for the CRs and the move to the new bundles format.

Now, I want to get this PR finished, test it on my own machine and not have to go back and forth with a  MILLIMETRE_PER_HOUR while I start implementing the auto-discovery feature.

I'd love to write my usual `100.millimeter / hour` (literal Scala syntax), but it took me at least an hour to figure out the proper syntax for `public static final Unit<Speed> MM_PER_HOUR = Units.METRE.divide(1000).divide(Units.HOUR).asType(Speed.class); `

So, it would be nice if you can accept the PR without this change.

Reply:
openHAB allowed Groovy once and it took ages to get rid of it again. Fortunately java gets more and more relaxed with newer versions. In Java 11 you can use "var" instead of repeating the type once more.

Regarding approval. I'm sure that @Hilbrand will not block the merge because of this one line, he just wants to reduce technical debts that accumulate if we merge code that is already "outdated". At the same time contributors like you cannot be expected to develop against OH core master, you are on the safe site.

Reply:
Ok, sorry, I'm reading my comment again and it reads like a rant.

(p.s. I'm not claiming to use Scala. I would write my own system from scratch in Scala if I wanted that. Still, I don't think one should compare Groovy with Scala, they are fundamentally different wrt type-safety.)

Reply:
No I won't block :smile: I do feel your pain. When I started I was asked twice: 'o btw can you use this new api instead' only to find out it wasn't complete and I had to add pr's to core. So can you at least leave a comment as suggested and use the same naming for the variable?